pkg/syncer/util.go

```diff
@@ -170,29 +171,29 @@ func getPVCKey(ctx context.Context, obj interface{}) (string, error) {
 }
 
 // HasMigratedToAnnotationUpdate returns true if the migrated-to annotation is found in the newer object
-func HasMigratedToAnnotationUpdate(ctx context.Context, prevAnnotations map[string]string, newAnnotations map[string]string) bool {
+func HasMigratedToAnnotationUpdate(ctx context.Context, prevAnnotations map[string]string, newAnnotations map[string]string, objectName string) bool {
 	log := logger.GetLogger(ctx)
 	// Checking if the migrated-to annotation is found in the newer object
 	if _, annMigratedToFound := newAnnotations[common.AnnMigratedTo]; annMigratedToFound {
 		if _, annMigratedToFound = prevAnnotations[common.AnnMigratedTo]; !annMigratedToFound {
-			log.Debugf("Received %v annotation update", common.AnnMigratedTo)
+			log.Debugf("Received %v annotation update for %s", common.AnnMigratedTo, objectName)
 			return true
 		}
 	}
-	log.Debugf("%v annotation not found", common.AnnMigratedTo)
+	log.Debugf("%v annotation not found for %s", common.AnnMigratedTo, objectName)
 	return false
 }
 
 // migratedToAnnotationExists returns true if the migrated-to annotation is found and equal to csi.vsphere.vmware.com
-func migratedToAnnotationExists(ctx context.Context, annotations map[string]string) bool {
+func migratedToAnnotationExists(ctx context.Context, annotations map[string]string, objectName string) bool {
```

Comment:
Instead of objectName and annotations, Can you supply pointer to  *metav1.ObjectMeta?

Reply:
Can we also refactor this function to check annotation contains `volume.beta.kubernetes.io/storage-provisioner: csi.vsphere.vmware.com`.

so you do not need multiple if checks at the caller side and it will make the code more readable.

Basically function should determine based supplied object metadata if PVC or PV is representing in-tree vSphere volume or not?

Reply:
As discussed can you add unit test cases to cover the following cases?

- `pv.kubernetes.io/migrated-to: csi.vsphere.vmware.com` annotation present - return `true`
- `pv.kubernetes.io/migrated-to: csi.vsphere.vmware.com` annotation not present, but `volume.beta.kubernetes.io/storage-provisioner: csi.vsphere.vmware.com` annotation is present - return `true`
- `pv.kubernetes.io/migrated-to: csi.vsphere.vmware.com` annotation not present and `volume.beta.kubernetes.io/storage-provisioner: csi.vsphere.vmware.com` annotation not present - return `false`